test/test-notification-model-updates.js

```diff
@@ -0,0 +1,382 @@
+
+      id: 0,
+      pid: 1234,
+      ppid: 3456,
+    };
+    app.handleModelUpdate(1, notification, function(err) {
+      tt.ok(!err);
+      tt.end();
+    });
+  });
+
+  t.test('Check instance data', function(tt) {
+    app.models.ServiceInstance.find({}, function(err, instances) {
+      tt.ok(!err, 'instances should be found');
+      tt.equal(instances.length, 1, 'only one instance should be found');
+      var inst = instances[0] || {};
+      tt.equal(inst.currentDeploymentId, 'some hash',
+        'commit hash should match');
+      tt.ok(!!inst.startTime, 'start time should be set');
+      tt.ok(!!inst.started, 'started should be true');
+      tt.equal(inst.applicationName, 'my app', 'app name should match');
+      tt.deepEqual(inst.npmModules, {name: 'list of npm modules'});
+      tt.equal(inst.PMPort, 5000, 'PM port should match');
+      tt.deepEqual(inst.containerVersionInfo, {
+        os: {
+          platform: os.platform(),
+          arch: os.arch(),
+          release: os.release()
+        },
+        node: process.version,
+        container: {
+          name: 'test-container',
+          version: 10.0
+        }
+      }, 'Container information should be set');
+      tt.equal(inst.setSize, 1, 'set size should match');
+      tt.equal(inst.agentVersion, '1.0.2', 'agent version should match');
+      tt.equal(inst.restartCount, 10, 'restart count should match');
+      tt.end();
+    });
+  });
+
+  t.test('Check worker 0', function(tt) {
+    var q = {where: {
+      workerId: 0,
+      serviceInstanceId: 1
+    }};
+    app.models.ServiceProcess.find(q, function(err, procs) {
+      tt.ok(!err, 'process should be found');
+      tt.equal(procs.length, 1, 'only one process should be found');
+      var proc = procs[0] || {};
+      tt.equal(proc.parentPid, 3456, 'parent pid should match');
+      tt.equal(proc.pid, 1234, 'pid should match');
+      tt.ok(!proc.isTrackingObjects, 'tracking-obj should be false');
+      tt.ok(!proc.isProfiling, 'tracking-obj should be false');
+      tt.ok(!proc.isSnapshotting, 'tracking-obj should be false');
+      tt.ok(!!proc.startTime, 'start time should be set');
+      tt.ok(!proc.stopTime, 'stop time should not be set');
+      tt.ok(!proc.stopReason, 'stop reason should not be set');
+      tt.end();
+    });
+  });
+
+  t.test('Notify fork', function(tt) {
+    var notification = {
+      cmd: 'fork',
+      id: 1,
+      pid: 1235,
+      ppid: 1234,
+    };
+    app.handleModelUpdate(1, notification, function(err) {
+      tt.ok(!err);
+      tt.end();
+    });
+  });
+
+  t.test('Notify listen', function(tt) {
+    var notification = {
+      cmd: 'listening',
+      id: 1,
+      pid: 1235,
+      address: {
+        address: '0.0.0.0',
+        port: 59176,
+        addressType: 4
+      }
+    };
+    app.handleModelUpdate(1, notification, function(err) {
+      tt.ok(!err);
+      tt.end();
+    });
+  });
+
+  function checkWorker1(prof, tt) {
```

Comment:
Understandable. I'm probably just flinching because of the pain caused by strong-pm's test/test-pmctl-local.js ;-)